network.go

```diff
@@ -678,7 +679,22 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 		}
 	}
 
-	if err = ep.assignAddress(true, !n.postIPv6); err != nil {
+	ipam, err := n.getController().getIPAM(n.ipamType)
+	if err != nil {
+		return nil, err
+	}
+
+	if ipam.capability.RequiresMAC {
+		if ep.iface.mac == nil {
+			ep.iface.mac = netutils.GenerateRandomMAC()
```

Comment:
From my understanding, with this change, the endpoint will get a random mac for DHCP based IPAM drivers.
But for other IPAM drivers (such as built-in), the Mac-address will be derived from ip-address. Correct ?
Not sure if this has any significance. But atleast we may have to document this somewhere ?